packages/sheets-ui/src/facade/f-univer.ts

```diff
@@ -86,6 +89,177 @@ export interface IFUniverSheetsUIMixin {
 }
 
 export class FUniverSheetsUIMixin extends FUniver implements IFUniverSheetsUIMixin {
+    override _initialize(injector: Injector): void {
+        const commandService = injector.get(ICommandService);
+        this.disposeWithMe(commandService.beforeCommandExecuted((commandInfo) => {
+            switch (commandInfo.id) {
+                case CopyCommand.id:
+                case CutCommand.id:
+                    this._beforeClipboardChange();
+                    break;
+                case SheetPasteShortKeyCommand.id:
+                    this._beforeClipboardPaste(commandInfo.params);
+                    break;
+            }
+        }));
+        this.disposeWithMe(commandService.onCommandExecuted((commandInfo) => {
+            switch (commandInfo.id) {
+                case CopyCommand.id:
+                case CutCommand.id:
+                    this._clipboardChanged();
+                    break;
+                case SheetPasteShortKeyCommand.id:
+                    this._clipboardPaste();
+                    break;
+            }
+        }));
+        // async listeners
+        this.disposeWithMe(commandService.beforeCommandExecuted(async (commandInfo) => {
+            switch (commandInfo.id) {
+                case PasteCommand.id:
+                    this._beforeClipboardPasteAsync();
+                    break;
+            }
+        }));
+        this.disposeWithMe(commandService.onCommandExecuted(async (commandInfo) => {
+            switch (commandInfo.id) {
+                case PasteCommand.id:
+                    this._clipboardPasteAsync();
+                    break;
+            }
+        }));
+    }
+
+    private _generateClipboardCopyParam(): IBeforeClipboardChangeParam | undefined {
+        const workbook = this.getActiveUniverSheet();
```

Comment:
We should encapsulate these lines like `getSheetCommandTarget`. They repeat lot of times in the codebase. cc @weird94